src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/ExecutorFactory.cs

```diff
@@ -33,8 +35,10 @@ public static class ExecutorFactory
                 var arguments = new object[handler.Parameters.Length];
                 for (var i = 0; i < handler.Parameters.Length; i++)
                 {
+                    // TODO: This should be cached?
+                    var binder = page.PageContext.HttpContext.RequestServices.GetRequiredService<PageArgumentBinder>();
                     var parameter = handler.Parameters[i];
-                    arguments[i] = await page.Binder.BindModelAsync(
+                    arguments[i] = await binder.BindModelAsync(
```

Comment:
To this file, basically. The only other solution I can come up with is passing the binder into `CreateExecutor` from `PageActionInvokerProvider`. Considering `PageActionInvokerProvider` already has a ton of private readonly's I guess that's cleaner as it contains the DI to one spot.